static/js/admin/src/AbstractForm.js

```diff
@@ -8,6 +8,11 @@ define('AbstractForm', [ 'jquery', 'jquery-ui', 'underscore', 'ModalCreate',
 
 		AbstractForm.prototype.open = function(resultObject) {
 			var dialogBox = new ModalCreate(this.options);
+
+			if (this.options.showCurrentFilename) {
```

Comment:
Hmm, I'd rather avoid adding the new filename related code into this class. You should be able to work this functionality into `EditFilenameForm.js`, maybe override the `open` method to set the filename and then call the parent method.